osv/purl_helpers.py

```diff
@@ -18,23 +18,44 @@
 
 from packageurl import PackageURL
 
-PURL_ECOSYSTEMS = {
-    'Alpine': 'apk',
-    'Bitnami': 'bitnami',
-    'crates.io': 'cargo',
-    'Debian': 'deb',
-    'Go': 'golang',
-    'Hackage': 'hackage',
-    'Hex': 'hex',
-    'Maven': 'maven',
-    'npm': 'npm',
-    'NuGet': 'nuget',
-    'OSS-Fuzz': 'generic',
-    'Packagist': 'composer',
-    'Pub': 'pub',
-    'PyPI': 'pypi',
-    'RubyGems': 'gem',
-    'SwiftURL': 'swift',
+# PURL spec:
+# https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst
+ECOSYSTEM_PURL_DATA = {
```

Comment:
PackageURL can't be used here because `purl.name` is a required field. Also I think PackageURL might not be the best fit since we are representing type and namespace per ecosystem. Replaced with a namedtuple.